### DISCUSSION TITLE
OSC to Keybind or Keybind to OSC

### DISCUSSION THREAD

**Discussion by Salvesti** [vrchat-community/osc#87] [Show and tell]
Here is a link to a github for my OSC to hotkey or hotkey to OSC program.
https://github.com/Salvesti/OSC_To_keypress_Console/

It lets you define your own hotkeys triggered by parameters, and allow for key combos to trigger OSCMessages into VRChat as well.

The readme has extra info in it, and the two files (which are mandatory to be in the directory you store the .exe in) ReceiverHotkeys.txt and SenderHotkeys.txt have some example lines in them.
You will need to modify these to have whatever parameters and key combos you want.

Either file can be empty as well if you don't want either aspect to run.

Unfortunately the internal codes that it uses to define the keys are a bit odd, so I included 'InfoForKeys.txt' which has the key names that the two hotkey files are looking for.
Some oddities are: 
ALT, which is defined as MENU.
Letter keys are VK_{Letter} so 'VK_A' for 'A'

I tried to make it relatively user friendly to modify the key binds.

**Comment by momo-the-monster** [comment]
@Salvesti - will you add a license to this project so it's clear what people can do with it?

**Reply by Salvesti** [reply]
I've added the MIT License to it now.
